### PR TITLE
Added Custom Navigation Example for iOS App

### PR DESCRIPTION
**Pull request by ttypic**
https://user-images.githubusercontent.com/16488867/221802265-7a5a492f-3583-495c-983f-f70c1f70e738.mp4

### CODE REVIEW

File: sample/app-ios/app-ios/CustomNavigationView.swift [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            ChildItems(component, observableChildren.value)
+            VStack {
+                Spacer()
+                Buttons(component, mode).padding()
+            }
+        }
+    }
+}
+
+extension CustomNavigationView {
```
Comment (arkivanov) [inline sample/app-ios/app-ios/CustomNavigationView.swift]: Why do we need this extension? Can the underlying structs just be top-level and private?

File: sample/app-ios/app-ios/CustomNavigationView.swift [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+    struct ChildItems: View {
+        private let component: CustomNavigationComponent
```
Comment (arkivanov) [inline sample/app-ios/app-ios/CustomNavigationView.swift]: This property looks unused?

File: sample/app-ios/app-ios/KittenView.swift [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                        ))
+                        .foregroundColor(.white)
+                        .modifier(AnimatableFontModifier(size: fontSize))
+                Spacer()
+            }
+        }
+    }
+}
+
+extension KittenView {
```
Comment (arkivanov) [inline sample/app-ios/app-ios/KittenView.swift]: The same question here. Maybe put the content as top-level and private?

**Comment by arkivanov** [comment]
This looks super cool! Thanks!

File: sample/app-ios/app-ios/CustomNavigationView.swift (hunk 2 above)
Comment (ttypic) [reply sample/app-ios/app-ios/CustomNavigationView.swift]: Deleted it, thanks!

File: sample/app-ios/app-ios/CustomNavigationView.swift (hunk 1 above)
Comment (ttypic) [reply sample/app-ios/app-ios/CustomNavigationView.swift]: Agree, made them private and top-level

File: sample/app-ios/app-ios/KittenView.swift (hunk 3 above)
Comment (ttypic) [reply sample/app-ios/app-ios/KittenView.swift]: Agree, made them private and top-level

**Comment by arkivanov** [comment]
Do you use Twitter? I would like to tweet about this and tag you. :-)

**Comment by ttypic** [comment]
I am not using Twitter much and I've just found out my account has been suspended (don't know why). But thank you very much for asking! I am really appreciate it.

**Comment by arkivanov** [comment]
Thanks!

**Pull request merged**